test/EndToEnd/tests/PackageNameSpaceTests.ps1

```diff
@@ -0,0 +1,171 @@
+function Test-PackageNameSpaceRestore-Succeed
+{
+    param($context)
+
+    # Arrange
+    $repoDirectory = $context.RepositoryRoot
+    $nugetConfigPath = Join-Path $OutputPath 'nuget.config'
+
+    $settingFileContent =@"
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+    <packageSources>
+    <clear />
+    <add key="ReadyPackages" value="{0}" />
+    </packageSources>
+    <packageNamespaces>
+        <packageSource key="ReadyPackages">
+            <namespace id="Soluti*" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>
+"@
+  
+    try {
+        # We have to create config file before creating solution, otherwise it's not effective for new solutions.
+        $settingFileContent -f $repoDirectory | Out-File -Encoding "UTF8" $nugetConfigPath
+    
+        $p = New-ConsoleApplication
+    
+        $projectDirectoryPath = $p.Properties.Item("FullPath").Value
+        $packagesConfigPath = Join-Path $projectDirectoryPath 'packages.config'
+        $projectDirectoryPath = $p.Properties.Item("FullPath").Value
+        $solutionDirectory = Split-Path -Path $projectDirectoryPath -Parent   
+        # Write a file to disk, but do not add it to project
+        '<packages>
+            <package id="SolutionLevelPkg" version="1.0.0" targetFramework="net461" />
+    </packages>' | out-file $packagesConfigPath
+    
+        # Act
+        Build-Solution
+    
+        # Assert
+        $packagesFolder = Join-Path $solutionDirectory "packages"
+        $solutionLevelPkgNupkgFolder = Join-Path $packagesFolder "SolutionLevelPkg.1.0.0"
+        Assert-PathExists(Join-Path $solutionLevelPkgNupkgFolder "SolutionLevelPkg.1.0.0.nupkg")
+        
+        $errorlist = Get-Errors
+        Assert-AreEqual 0 $errorlist.Count
+    }
+    finally {
+        Remove-Item $nugetConfigPath
+    }
+}
+
+function Test-PackageNameSpaceIdenticalPackageIdsRestore-CorrectPackage
```

Comment:
Consider

Test-PackageNamespaceRestore-WithMultipleFeedsWithIdenticalPackages-RestoresCorrectPackage

Reply:
Make the names more explicit about what they are testing, even if the name is longer.